cai_causal_graph/graph_components.py

```diff
@@ -193,12 +202,22 @@ def to_dict(self, include_meta: bool = True) -> dict:
         node_dict = {
             'identifier': self.identifier,
             'variable_type': self.variable_type,
+            'node_class': self.__class__.__name__,
         }
         if include_meta:
             node_dict['meta'] = deepcopy(self.meta)   # type: ignore
 
         return node_dict
 
+    @classmethod
+    def from_dict(cls, node_dict: dict) -> Node:
+        """Return a `cai_causal_graph.graph_components.Node` instance from a dictionary."""
```

Comment:
Would add the `assert 'identifier' in node_dict` just to be consistent with `TimeSeriesNode` as so user gets a nice message and not just a `KeyError`.